crates/bevy_reflect/src/impls/std.rs

```diff
@@ -394,6 +394,12 @@ impl<K: FromReflect + Eq + Hash, V: FromReflect> Map for HashMap<K, V> {
         self.insert(key, value)
             .map(|old_value| Box::new(old_value) as Box<dyn Reflect>)
     }
+
+    fn remove(&mut self, key: &dyn Reflect) -> Option<Box<dyn Reflect>> {
+        key.downcast_ref::<K>()
```

Comment:
This will fail if `key` is _any_ Dynamic type. It would be safer to use `FromReflect` if the downcast fails.

Honestly, `get` and `get_mut` should do this as well, but that can be addressed in a separate PR.